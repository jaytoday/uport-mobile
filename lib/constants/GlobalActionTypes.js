// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
export const RESET_DEVICE = 'RESET_DEVICE'
export const RELOAD_DB = 'RELOAD_DB'
export const LOADED_DB = 'LOADED_DB'
export const PERSIST_DB = 'PERSIST_DB'
export const FORCE_PERSIST_DB = 'FORCE_PERSIST_DB'
