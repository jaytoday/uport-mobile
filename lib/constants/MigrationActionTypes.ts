import MigrationScreen from '../components/Migrations/MigrationScreen'

//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

export const ADD_MIGRATION_TARGET = 'ADD_MIGRATION_TARGET'
export const RUN_MIGRATIONS = 'RUN_MIGRATIONS'
export const STARTED_MIGRATION_STEP = 'STARTED_MIGRATION_STEP'
export const COMPLETED_MIGRATION_STEP = 'COMPLETED_MIGRATION_STEP'
export const FAILED_MIGRATION_STEP = 'FAILED_MIGRATION_STEP'

import SCREENS from 'uPortMobile/lib/screens/Screens'

export enum MigrationTarget {
  Legacy = 'Legacy',
  MissingKeys = 'MissingKeys',
  RecoverSeed = 'RecoverSeed',
}

export enum MigrationStep {
  MigrateLegacy = 'MigrateLegacy',
  EnterSeed = 'EnterSeed',
}

export enum MigrationStatus {
  NotStarted,
  Started,
  Completed,
  Error,
}

export interface Recipes {
  [index: string]: MigrationStep[]
}

export const targetRecipes: Recipes = {
  Legacy: [MigrationStep.MigrateLegacy],
  RecoverSeed: [MigrationStep.EnterSeed],
}

export interface MigrationScreens {
  [index: string]: string
}

export const migrationScreens: MigrationScreens = {
  RecoverSeed: SCREENS.RECOVERY.RestoreSeedInstructions,
}

export interface TargetAction {
  type: 'ADD_MIGRATION_TARGET' | 'RUN_MIGRATIONS'
  target: MigrationTarget
  _backup?: boolean
}

export interface StepAction {
  type: 'STARTED_MIGRATION_STEP' | 'COMPLETED_MIGRATION_STEP' | 'FAILED_MIGRATION_STEP'
  step: MigrationStep
  _backup?: boolean
}
