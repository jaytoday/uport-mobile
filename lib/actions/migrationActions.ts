//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//

import { ADD_MIGRATION_TARGET, STARTED_MIGRATION_STEP, COMPLETED_MIGRATION_STEP, FAILED_MIGRATION_STEP, RUN_MIGRATIONS, MigrationStep, MigrationTarget, TargetAction, StepAction } from 'uPortMobile/lib/constants/MigrationActionTypes'

const _backup = true

export function addMigrationTarget(target: MigrationTarget) : TargetAction {
  return {
    type: ADD_MIGRATION_TARGET,
    target,
    _backup
  }
}

export function runMigrations(target: MigrationTarget) : TargetAction {
  return {
    type: RUN_MIGRATIONS,
    target
  }
}

export function startedMigrationStep(step: MigrationStep) : StepAction {
  return {
    type: STARTED_MIGRATION_STEP,
    step,
    _backup
  }
}

export function completedMigrationStep(step: MigrationStep) : StepAction {
  return {
    type: COMPLETED_MIGRATION_STEP,
    step,
    _backup
  }
}

export function failedMigrationStep(step: MigrationStep) : StepAction {
  return {
    type: FAILED_MIGRATION_STEP,
    step,
    _backup
  }
}