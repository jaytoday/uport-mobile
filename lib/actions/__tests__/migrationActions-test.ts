// This file is part of the uPort Mobile App.

// The uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.

// The uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.

// You should have received a copy of the GNU General Public License
// along with The uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.

// (C) Copyright 2016-2018 ConsenSys AG

import { addMigrationTarget, runMigrations, startedMigrationStep, completedMigrationStep, failedMigrationStep } from '../migrationActions'
import { MigrationStep, MigrationTarget } from 'uPortMobile/lib/constants/MigrationActionTypes'

it('creates a ADD_MIGRATION_TARGET action', () => {
  expect(addMigrationTarget(MigrationTarget.Legacy)).toMatchSnapshot()
})

it('creates a RUN_MIGRATIONS action', () => {
  expect(runMigrations(MigrationTarget.Legacy)).toMatchSnapshot()
})

it('creates a STARTED_MIGRATION_STEP action', () => {
  expect(startedMigrationStep(MigrationStep.MigrateLegacy)).toMatchSnapshot()
})

it('creates a COMPLETED_MIGRATION_STEP action', () => {
  expect(completedMigrationStep(MigrationStep.MigrateLegacy)).toMatchSnapshot()
})

it('creates a FAILED_MIGRATION_STEP action', () => {
  expect(failedMigrationStep(MigrationStep.MigrateLegacy)).toMatchSnapshot()
})
