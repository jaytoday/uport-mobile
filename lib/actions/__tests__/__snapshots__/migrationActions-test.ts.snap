// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates a ADD_MIGRATION_TARGET action 1`] = `
Object {
  "_backup": true,
  "target": "Legacy",
  "type": "ADD_MIGRATION_TARGET",
}
`;

exports[`creates a COMPLETED_MIGRATION_STEP action 1`] = `
Object {
  "_backup": true,
  "step": "MigrateLegacy",
  "type": "COMPLETED_MIGRATION_STEP",
}
`;

exports[`creates a FAILED_MIGRATION_STEP action 1`] = `
Object {
  "_backup": true,
  "step": "MigrateLegacy",
  "type": "FAILED_MIGRATION_STEP",
}
`;

exports[`creates a RUN_MIGRATIONS action 1`] = `
Object {
  "target": "Legacy",
  "type": "RUN_MIGRATIONS",
}
`;

exports[`creates a STARTED_MIGRATION_STEP action 1`] = `
Object {
  "_backup": true,
  "step": "MigrateLegacy",
  "type": "STARTED_MIGRATION_STEP",
}
`;
